core/router/gateway/gateway.go

```diff
@@ -26,4 +31,71 @@ type Gateway struct {
 	Utilization Utilization
 	Schedule    Schedule
 	LastSeen    time.Time
+	Token       string
+
+	monitor *monitorConn
+
+	Ctx log.Interface
+}
+
+func (g *Gateway) updateTimestamp() {
+	g.LastSeen = time.Now()
+}
+
+func (g *Gateway) HandleStatus(status *pb.Status) (err error) {
```

Comment:
We probably want to do this after we get the result, because at this point, the func might still fail